packages/blade/src/components/List/ListContext.ts

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import type { IconComponent } from '../Icons';
+import type { ListProps } from './List';
+
+export type ListContextType = {
+  level?: number;
+  size: NonNullable<ListProps['size']>;
+  icon?: IconComponent;
+  variant: NonNullable<ListProps['variant']>;
+};
+
+const ListContext = React.createContext<ListContextType>({
+  level: undefined,
```

Comment:
So List will be used within another List which means that the ListContext will exist within another ListContext. So by default, the level of any ListContext is unknown (undefined) until its evaluated by checking if another ListContext exists. So only on level 1 will we have the default `undefined` level but as we move deeper the level will be defined. So for the convenience of doing `level = listContext.level ? listContext.level +1 : 1` we set it to be undefined before any evaluation

Reply:
then make this as 0th level instead of undefined? at least it will be predictable?

Reply:
Its not sitting well with the logic. List can be of level 1, 2, 3 or it can be unknown. It will never be a level 0. Making it 0 will mean that I will have to do something like thi
```diff
- level: level ? level + 1 : 1
+ level: level +1 
```

Now looking at this it would make me think that level is _always_ incremented even when there is just 1 list which looks like incorrect logic. Whereas doing a check if the level is unknown then make it 1 otherwise increment it seems more logical.